stock_kardex_report/wizard/stock_kardex_report_wizard.py

```diff
@@ -0,0 +1,144 @@
+    _description = 'Wizard to create kardex reports of stock moves'
+
+    date_from = fields.Datetime(
+        string='From', required=True, default=fields.Datetime.now)
+    date_to = fields.Datetime(
+        string='To', required=True, default=fields.Datetime.now)
+    product = fields.Many2one('product.product', required=True)
+    location = fields.Many2one('stock.location', required=True)
+
+    def open_table(self):
+        self.env['stock.kardex.report'].search([]).unlink()
+        self._cr.execute('''
+        SELECT
+        a.done - b.done
+        AS
+        total
+        FROM
+        (
+            SELECT sum(qty_done)
+            AS
+            done
+            FROM
+            stock_move_line
+            WHERE
+            product_id = %s
+            AND
+            state = \'done\'
+            AND
+            date < %s
+            AND
+            location_dest_id = %s
+        )
+        a
+        CROSS JOIN
+        (
+            SELECT sum(qty_done)
+            AS
+            done
+            FROM
+            stock_move_line
+            WHERE
+            product_id = %s
+            AND
+            state = \'done\'
+            AND
+            date < %s
+            AND
+            location_id = %s
+        )
+        b
+        ''', [
+            self.product.id, self.date_from, self.location.id,
+            self.product.id, self.date_from, self.location.id
+            ])
+        start_qty = self._cr.dictfetchall()
+        total = start_qty[0]['total']
+        self._cr.execute("""WITH one AS (
+            SELECT
+            sml.product_id, sml.product_uom_id,
+            sml.lot_id, sml.owner_id, sml.package_id,
+            sml.qty_done, sml.move_id, sml.location_id,
+            sml.location_dest_id, sm.date, sm.origin,
+            sm.state
+            FROM stock_move_line sml
+            INNER JOIN stock_move sm
+            ON sml.move_id = sm.id
+            WHERE
+            sm.date >= %s
+            AND sm.date <= %s),
+            two AS (
+                SELECT *
+                FROM one
+                WHERE location_id = %s
+                OR location_dest_id = %s)
+            SELECT *
+            FROM two
+            WHERE product_id = %s
+            AND state = 'done'
+            ORDER BY date;""", [
+            self.date_from, self.date_to,
+            self.location.id, self.location.id,
+            self.product.id
+            ])
+        moves = self._cr.dictfetchall()
+        report_list = []
+        report_list.append({
+            'move_id': False,
+            'product_id': self.product.id,
+            'product_uom_id': False,
+            'lot_id': False,
+            'owner_id': False,
+            'package_id': False,
+            'qty_done': 0,
+            'location_id': False,
+            'location_dest_id': False,
+            'date': self.date_from,
+            'origin': 'Initial Balance',
+            'balance': total,
+        })
```

Comment:
```suggestion
        report_list.append({
            'product_id': self.product.id,
            'product_uom_id': product_id.uom_id.id,
            'date': self.date_from,
            'origin': _('Initial Balance'),
            'balance': total,
        })
```

If you don't use the fields you don't need to define it as False